templates/native/react-redux/test/template.reducer.spec.js

```diff
@@ -76,4 +76,22 @@ describe('TEMPLATE_LOWER_CAMEL_CASE_NAMEReducer - Unit Test', () => {
     each of the asynchronous action's scenarios in its own 'it'. Remember to test happy and sad cases.
     If you have no asynchronous actions, this code may be omitted.
     */
+    describe('exampleAsyncAction', () => {
+        it('should dispatch exampleAction', () => {
+            /*
+             Mock dispatch and getState
+             */
+            const getState = jest.fn(() => ({
```

Comment:
```
const getState = jest
    .fn()
    .mockImplementation(() => ({
        TEMPLATE_LOWER_CAMEL_CASE_NAMEReducer: { exampleVariable: false }
    }))
    .mockName('getState');
    const dispatch = jest.fn().mockName('dispatch');
```

Reply:
I quite like jest's new `mockImplimentation` and `mockName` tags. 
When errors occur it doesn't just say `expect jest.fn() to have been called with...`, it'll give you the mock name instead. 
And when you want to mock out a different implementation of a method for a specific test, you can go `method.mockImplimentation(() => someNewFunction())` without having to re-mock and rename the original mocked method.

Reply:
eish didnt have internet today at work. I like this better. I'll make the change

Reply:
@JPStrydom I'm getting errors with above code. I think we might introducing breaking changes with `mockName`.

Reply:
CRA uses a *VERY* old version of `Jest` - but you're right, we better not introduce breaking features (as `mockName` is quite new).

I was also wondering, what do you think of using

```
expect(dispatch.mock.calls[0][0]).toEqual(x(y));
```

as opposed to

```
expect(dispatch).toHaveBeenCalledWith(x(y));
```

The latter might be a little less confusing for new users - and it's easier to add more checks. Only downside is that is doesn't ensure the call order.

Reply:
I like `expect(disptach).toHaveBeenCalledWith` or though we loose call order I think you are right more people will like this version. we can change it to use this.

So from jest 23 there's `.toHaveBeenNthCalledWith(nthCall, arg1, arg2, ....)` which gives you call order check. Therefore in future, we can change to this.

http://jestjs.io/docs/en/expect.html#tohavebeennthcalledwithnthcall-arg1-arg2-

Reply:
And from `Jest` 23 we can use `mockName` ☺️

I like it!

Reply:
Perhaps we can add a comment mentioning that if users have `jest >= 23.x` then they can use the nth call check method if they want.